### PR TITLE
add extension support to the CRLBuilder

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #2559 and #2533

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -1510,7 +1530,27 @@ def create_x509_crl(self, builder, private_key, algorithm):
...
+            extension = self._lib.X509_EXTENSION_create_by_OBJ(
+                self._ffi.NULL,
+                obj,
+                1 if extension.critical else 0,
+                _encode_asn1_str_gc(self, pp[0], r)
+            )
+            self.openssl_assert(extension != self._ffi.NULL)
+            extension = self._ffi.gc(extension, self._lib.X509_EXTENSION_free)
+            res = self._lib.X509_CRL_add_ext(x509_crl, extension, i)
+            self.openssl_assert(res == 1)
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Ok, we already have this loop 2x, I hereby declare it to be time to write a function that does all the things the same way all the times.

Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: (That's me asking for a separate PR which refactors the existing two functions into a reusable thang)

Comment (reaperhulk) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Hehe, I was going to do that as the next PR after this merged, but I can do it first :)

File: tests/test_x509_crlbuilder.py [hunk 2]
```diff
@@ -144,6 +160,108 @@ def test_sign_empty_list(self, backend):
...
+        ).next_update(
+            next_update
+        ).add_extension(
+            extension, False
+        )
+
+        crl = builder.sign(private_key, hashes.SHA256(), backend)
+        assert len(crl) == 0
+        assert len(crl.extensions) == 1
+        ext = crl.extensions.get_extension_for_class(extension.__class__)
```
Comment (alex) [inline tests/test_x509_crlbuilder.py]: Prefer `type(exetension)` here.

**Pull request merged**